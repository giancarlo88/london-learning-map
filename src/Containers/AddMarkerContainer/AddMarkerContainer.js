import React, { Component } from 'react'
import LoginForm from '../../Components/LoginForm/LoginForm'
import AddMarkerForm from '../../Components/AddMarkerForm/AddMarkerForm'
import { post } from '../../Services/api'

class AddMarkerContainer extends Component {
  constructor() {
    super()
    this.state = {
      isAuthenticated: false
    }
  }

  handleLogonFormSubmit = async e => {
    e.preventDefault()
    const body = {
      username: 'gcgp88',
      password: 'asdf1234'
    }
    try {
      const res = await post(`${process.env.REACT_APP_ENDPOINT_URL}/auth`, body)
      if (res && res.status === 200) {
        this.setState({
          isAuthenticated: true
        })
      }
    } catch (err) {
      console.error(err)
    }
  }

  render() {
    const { isAuthenticated } = this.state
    return isAuthenticated ? (
      <AddMarkerForm />
    ) : (
      <LoginForm handleSubmit={this.handleLogonFormSubmit} />
    )
  }
}

export default AddMarkerContainer
