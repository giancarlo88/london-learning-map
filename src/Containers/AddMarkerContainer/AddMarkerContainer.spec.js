import React from 'react'
import { shallow } from 'enzyme'
import AddMarkerContainer from './AddMarkerContainer'

describe('AddMarkerContainer', () => {
  describe('@renders', () => {
    describe('with default conditions', () => {
      const wrapper = shallow(<AddMarkerContainer />)
      
      it('should render', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('should render a LoginForm', () => {
        expect(wrapper.find('LoginForm')).toHaveLength(1)
      });
    })

    describe('when state property isAuthenticated is true', () => {
      const wrapper = shallow(<AddMarkerContainer />)
      wrapper.setState({isAuthenticated: true})
      it('should render a AddMarkerForm', () => {
        expect(wrapper.find('AddMarkerForm')).toHaveLength(1)
      });
    });

  })
})
