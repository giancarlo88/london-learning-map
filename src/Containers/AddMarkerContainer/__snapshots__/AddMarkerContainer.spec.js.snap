// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMarkerContainer @renders with default conditions should render 1`] = `
<LoginForm
  handleSubmit={[Function]}
/>
`;
